src/components/NodesTable.tsx

```diff
@@ -292,42 +286,27 @@ const PendingIndicatorRoot = styled.button`
 type SubmitNodeAddressesCallback = (
     operatorId: string,
     addresses: string[],
-    options?: { onSuccess?: () => void; onError?: (e: unknown) => void },
+    options?: {
+        onSuccess?: (blockNumber: number) => void
+        onError?: (e: unknown) => void
+    },
 ) => Promise<void>
 
 export function useSubmitNodeAddressesCallback(): [SubmitNodeAddressesCallback, boolean] {
     const [isSubmitting, setIsSubmitting] = useState(false)
 
-    const abortControllerRef = useRef<AbortController>()
-
-    useEffect(() => {
-        const { current: abortController } = abortControllerRef
-
-        return () => {
-            abortController?.abort()
-        }
-    }, [])
-
     const cb: SubmitNodeAddressesCallback = useCallback(
         async (operatorId, addresses, { onSuccess, onError } = {}) => {
-            abortControllerRef.current?.abort()
-
-            const abortController = new AbortController()
-
-            abortControllerRef.current = abortController
-
             setIsSubmitting(true)
 
-            const { signal } = abortController
-
             try {
-                await setOperatorNodeAddresses(operatorId, addresses)
-
-                if (!signal.aborted) {
-                    onSuccess?.()
-                }
+                await setOperatorNodeAddresses(operatorId, addresses, {
+                    onBlockNumber(blockNumber) {
+                        onSuccess?.(blockNumber)
+                    },
+                })
```

Comment:
I'm gonna progressively convert utils to support similar side effect (`onBlockNumber`). Bubbling it up will make things much more explicit.